src/domain/core/src/messages/core_message_types.rs

```diff
@@ -75,3 +109,13 @@ pub struct DatasetLifecycleMessageDeleted {
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DatasetLifecycleMessageRenamed {
+    pub dataset_id: DatasetID,
+    pub owner_account_id: AccountID,
+    pub old_dataset_name: DatasetName,
+    pub new_dataset_name: DatasetName,
+}
```

Comment:
Note:

> re `owner_account_id` & `old_dataset_name`

Previously, I was asked to add these fields (comment): https://github.com/kamu-data/kamu-cli/pull/761#discussion_r1730092925